packages/loot-core/src/server/budget/goaltemplates.ts

```diff
@@ -97,22 +91,27 @@ async function processTemplate(month, force, category_templates) {
             : lowestPriority;
       }
     }
-  }
-
-  await setZero({ month });
-
-  //setZero() sets budgeted Income to 0. Reset income categories before continuing.
-  if (isReflectBudget()) {
-    for (let l = 0; l < originalCategoryBalance.length; l++) {
-      if (originalCategoryBalance[l].isIncome) {
-        await setBudget({
-          category: originalCategoryBalance[l].cat.id,
-          month,
-          amount: originalCategoryBalance[l].amount,
-        });
-      }
+    if (budgeted) {
+      originalCategoryBalance.push({
+        category: category.id,
+        amount: budgeted,
+        isIncome: category.is_income,
+        isTemplate: template ? true : false,
+      });
+      setToZero.push({
```

Comment:
This is doing the same thing, but substituting the amount for 0 to only zero.  We're also recording whether or not it's an income category.